File: haskell/eg/chaser.hs
import Tankode.Basic

ident :: Id
ident = Id
  { name = "chaser"
  , trackColour  = "blue2"
  , bodyColour   = "blue4"
  , gunColour    = "grey7"
  , radarColour  = "grey1"
  , bulletColour = "grey9"
  , scanColour   = "blue1"
  }

chaser :: Tankode () -- :: Input () -> Output ()
chaser Input {enemy = Just d} = output
  { shoot = 1
  , accel = a
  }
  where
  a | d > 1     =  1
    | d < 1     = -1
    | otherwise =  0
chaser Input {enemy = Nothing, speed = s} = output
  { body = 1
  , accel = a
  }
  where
  a | s > 0     = -1
    | s < 0     =  1
    | otherwise =  0

main :: IO ()
main = run ident chaser ()

File: c/raw/eg/chaser.c
#include <tankode.h>

struct tankode_out chaser(struct tankode_in in)
{
	struct tankode_out out = {0., 0., 0., 0., 0.};
	if (in.scanned_enemy) {
		out.shoot = 1.;
		out.accel = in.enemy > 1 ? +1 : in.enemy < 1 ? -1 : 0;
	} else {
		out.body  = +1.;
		out.accel = in.speed > 0 ? -1 : in.speed < 0 ? +1 : 0;
	}
	return out;
}

int main()
{
	struct tankode_id id;
	id.name   = "chaser";
	id.track  = "blue2";
	id.body   = "blue4";
	id.gun    = "grey7";
	id.radar  = "grey1";
	id.bullet = "grey9";
	id.scan   = "blue1";
	tankode_run(id, chaser);
	return 0;
}
